### PR TITLE
Added Keys method that returns a list of stored keys.

### PR DESCRIPTION
**Pull request by alesstimec**

### CODE REVIEW

File: kv.go [hunk 1]
```diff
@@ -69,6 +69,9 @@ type Store interface {
 	// collected at some point after that time. Clients should not
 	// rely on the value being removed at the given time.
 	Update(ctx context.Context, key string, expire time.Time, getVal func(old []byte) ([]byte, error)) error
+
+	// Keys returns a distinct list of stored keys.
+	Keys(ctx context.Context) ([]string, error)
```
Comment (mhilton) [inline kv.go:74]: As this extends the interface that is designed to be implemented by external things I would suggest creating a new interface that includes this one and extends it with the Keys method.

File: sqlsimplekv/keyvalue.go [hunk 2]
```diff
@@ -153,6 +153,30 @@ func (s *kvStore) Update(ctx context.Context, key string, expire time.Time, getV
...
+		}
+		keys = append(keys, key)
+	}
+	if err := rows.Err(); err != nil {
+		return nil, errgo.Mask(err)
+	}
+	return keys, nil
+}
+
 type nullTime struct {
```
Comment (mhilton) [inline sqlsimplekv/keyvalue.go:180]: this could be replaces with sql.NullTime these days.

**Review by mhilton — APPROVED** [review]
LGTM with a few thoughts.

File: kv.go (hunk 1 above)
Comment (alesstimec) [reply kv.go:79]: done

**Review by mhilton — APPROVED** [review]

**Pull request merged**